### PR TITLE
QtCollider: fix method linking

### LINKED ISSUE #4933: WebView findText throws a primitive error
## Environment

* SuperCollider version: 3.11.0 (develop-1a728802c)
* Operating system: macOS 10.14.6
* Other details (Qt version, audio driver, etc.): Qt 5.14.1 (homebrew)

## Steps to reproduce

Calling the `findText` method triggers the error. The same error can be reproduced by opening a HelpBrowser instance and using the "Find the text in document" utility (Ctrl-f). 

```supercollider
(
w = Window();
v = WebView(w, Rect(0,0,w.bounds.width,w.bounds.height));
v.findText("foo");
)
```

## Expected vs. actual behavior

```
ERROR: Primitive '_QObject_InvokeMethod' failed.
Failed.
RECEIVER:
Instance of WebView {    (0x119a304d8, gc=74, fmt=00, flg=00, set=06)
  instance variables [49]
    qObject : RawPointer 0x7fdc0a0a3450
    finalizer : instance of Finalizer (0x11bafd6b8, size=2, set=1)
    virtualSlots : nil
    wasRemoved : false
    font : nil
    resize : Integer 1
    alpha : Float 1.000000   00000000 3FF00000
    decorator : nil
    layout : nil
    userCanClose : true
    deleteOnClose : true
    action : nil
    mouseDownAction : nil
    mouseUpAction : nil
    mouseEnterAction : nil
    mouseLeaveAction : nil
    mouseMoveAction : nil
    mouseOverAction : nil
    mouseWheelAction : nil
    keyDownAction : nil
    keyUpAction : nil
    keyModifiersChangedAction : nil
    keyTyped : nil
    focusGainedAction : nil
    focusLostAction : nil
    dragLabel : nil
    beginDragAction : nil
    canReceiveDragHandler : nil
    receiveDragHandler : nil
    toFrontAction : nil
    endFrontAction : nil
    onClose : nil
    onResize : nil
    onMove : nil
    onLoadFinished : nil
    onLoadFailed : nil
    onLoadProgress : nil
    onLoadStarted : nil
    onLinkActivated : nil
    onLinkHovered : nil
    onReloadTriggered : nil
    onJavaScriptMsg : nil
    onSelectionChanged : nil
    onTitleChanged : nil
    onUrlChanged : nil
    onScrollPositionChanged : nil
    onContentsSizeChanged : nil
    onAudioMutedChanged : nil
    onRecentlyAudibleChanged : nil
}
PATH: /private/tmp/foo.scd
CALL STACK:
	MethodError:reportError
		arg this = <instance of PrimitiveFailedError>
	Nil:handleError
		arg this = nil
		arg error = <instance of PrimitiveFailedError>
	Thread:handleError
		arg this = <instance of Thread>
		arg error = <instance of PrimitiveFailedError>
	Object:throw
		arg this = <instance of PrimitiveFailedError>
	Object:primitiveFailed
		arg this = <instance of WebView>
	WebView:findText
		arg this = <instance of WebView>
		arg text = "foo"
		arg reversed = false
		arg func = <instance of Function>
	Interpreter:interpretPrintCmdLine
		arg this = <instance of Interpreter>
		var res = nil
		var func = <instance of Function>
		var code = "(
w = Window();
v = WebView(..."
		var doc = <instance of Document>
		var ideClass = nil
	Process:interpretPrintCmdLine
		arg this = <instance of Main>
^^ The preceding error dump is for ERROR: Primitive '_QObject_InvokeMethod' failed.
Failed.
RECEIVER: a WebView
```

### PR DESCRIPTION
**Pull request by brianlheim**
## Purpose and Motivation

Fixes #4933.

We need to use the fully qualified "QtCollider::QcCallback", otherwise the qt metatype system
doesn't find the function signature. this seems to be a deficiency in the way qt matches function
signatures, but solving it more cleverly might be difficult and introduce other bugs, so i'm just
doing the straightforward thing for now.

another problem (maybe?) raised by this is that findText on a web view that isn't visible will
lock up the interpreter. same with findText(""). i don't know why.

## Types of changes

- Documentation
- Bug fix

## To-do list

- [x] Code is tested
- [x] All tests are passing
- [x] Updated documentation
- [x] This PR is ready for review

### CODE REVIEW

**Comment by brianlheim** [comment]
FYI this has been tested by the issue reporter and found to work: https://github.com/supercollider/supercollider/issues/4933#issuecomment-629789093

**Comment by dyfer** [comment]
Thanks @brianlheim 
The fix seems to be working.
Unfortunately, tests are hit-or-miss on my system, because of timing I assume (it worked fine when I increase wait times). I was only vaguely following conversations in the past about a) async behavior in tests b) gui timing in tests, but it seems to be an issue here. Is there any way to improve this?

File: testsuite/classlibrary/TestWebView.sc [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+		webView.findText("abc", true, { |found|
+			"found".postln;
+			cond.unhang;
+			wasFound = found;
+		});
+		timeout = fork {
+			0.1.wait;
+			cond.unhang;
+		};
+
```
Comment (dyfer) [inline testsuite/classlibrary/TestWebView.sc:21]: When I saved this file in scide I think this whitespace was automatically deleted... also in line 27.

**Comment by brianlheim** [comment]
thanks for the review @dyfer -- through some experimentation i found that the 'load finished' callback always fires when the html is updated, so i updated my tests to use that instead of a direct wait. also removed the whitespace, thanks for pointing that out.

**Comment by brianlheim** [comment]
should be ready for another review now!

**Comment by dyfer** [comment]
Thanks @brianlheim 

This is better, however the test still times out on my system, and when it does, I get an error:
```
FAIL: a TestWebView: test_findText_found - test timeout check

ERROR: Message 'not' not understood.
RECEIVER:
   nil
ARGS:
(...)
```
I believe this is because `wasFound` is still `nil`, maybe if could be initialized to `false`?
Also, would you consider increasing the timeout?

**Comment by brianlheim** [comment]
> maybe if could be initialized to false?

i specifically want `nil` to indicate a timeout. updated so these asserts are run conditionally

> Also, would you consider increasing the timeout?

sure, i updated it to 2 seconds

thanks @dyfer ! ready for more review.

**Pull request merged**